firmware/src/mPMTTestStand.cpp

```diff
@@ -111,34 +112,108 @@ void mPMTTestStand::handle_get_data()
     DataId data_id = (DataId)((this->comm.received_message().data)[0]);
     switch (data_id) {
         case DATA_MOTOR:
+        {
+            uint32_t axis_x_pos = this->x_state->encoder_current;
+            uint32_t axis_y_pos = this->y_state->encoder_current;
+
             uint8_t data[2*4];
-            HTONL(data, axis_x.encoder.current);
-            HTONL(data + 4, axis_y.encoder.current);
+            HTONL(data, axis_x_pos);
+            HTONL(data + 4, axis_y_pos);
             this->comm.data(data, sizeof(data));
             break;
-
+        }
         case DATA_TEMP:
+        {
             // TODO
             break;
+        }
     }
 }
 
+void mPMTTestStand::debug_dump_axis(AxisId axis_id)
+{
+    const AxisState *state = (axis_id == AXIS_X ? this->x_state : this->y_state);
+    DEBUG_PRINTLN("----------------------------------------");
+    DEBUG_PRINT_VAL("AXIS", axis_id == AXIS_X ? "X" : "Y");
+    DEBUG_PRINT_VAL("moving          ", state->moving);
+    DEBUG_PRINT_VAL("ls_home_pressed ", state->ls_home_pressed);
+    DEBUG_PRINT_VAL("ls_far_pressed  ", state->ls_far_pressed);
+    DEBUG_PRINT_VAL("velocity        ", state->velocity);
+    DEBUG_PRINT_VAL("next_velocity   ", state->next_velocity);
+    DEBUG_PRINT_VAL("velocity_segment", state->velocity_segment);
+    DEBUG_PRINT_VAL("encoder_current ", state->encoder_current);
+    DEBUG_PRINT_VAL("encoder_target  ", state->encoder_target);
+    DEBUG_PRINT_VAL("dir             ", state->dir);
+    DEBUG_PRINTLN("----------------------------------------");
+}
+
+void mPMTTestStand::debug_dump()
+{
+    DEBUG_PRINT_VAL("STATUS", this->status);
+    this->debug_dump_axis(AXIS_X);
+    this->debug_dump_axis(AXIS_Y);
+}
+
 void mPMTTestStand::execute()
 {
-    // Update status
-    Status old_status = this->status;
-    if (axis_x.moving || axis_y.moving) {
-        this->status = (axis_x.homing || axis_y.homing) ? STATUS_HOMING : STATUS_MOVING;
-    }
-    else if (axis_x.fault || axis_y.fault) {
-        this->status = STATUS_FAULT;
-    }
-    else {
-        this->status = STATUS_IDLE;
-    }
+    PERIODIC(this->debug_dump(), 1000);
 
-    if (this->status != old_status) {
-        DEBUG_PRINT_VAL("Status updated to", this->status);
+    // Update status
+    switch (this->status) {
+        case STATUS_IDLE:
+            break;
+        case STATUS_MOVING:
+            if (this->x_state->moving || this->y_state->moving) {
+                this->status = STATUS_MOVING;
+            }
+            else if (this->x_state->ls_home_pressed || this->x_state->ls_far_pressed ||
+                     this->y_state->ls_home_pressed || this->x_state->ls_far_pressed) {
+                this->status = STATUS_FAULT;
```

Comment:
Did we remove the fault status earlier on?

Reply:
It was removed, then re-added, but I forgot it in that other spot, see the comment above.